### PR TITLE
Add searching Functionality to the searchBar in the HomeView.

### PR DESCRIPTION
**Pull request by MohamedSaiko**
Requesting Code Review and any recommended updates.

Simulator Screen Shot - iPhone 13 Pro - 2024-03-25 at 13 57 20

Simulator Screen Shot - iPhone 13 Pro - 2024-03-25 at 13 57 36

Simulator Screen Shot - iPhone 13 Pro - 2024-03-25 at 13 57 45

Simulator Screen Shot - iPhone 13 Pro - 2024-03-25 at 13 57 27

### CODE REVIEW

File: iStore/Home/HomeView.swift [hunk 1]
```diff
@@ -29,13 +28,23 @@ struct HomeView: View {
                     ProductsGrid(products: homeViewModel.products, userID: authenticationViewModel.user.id)
                 }
             }
-            .searchable(text: $text)
             .padding(.horizontal)
             .navigationTitle(iStore)
             .navigationBarTitleDisplayMode(.inline)
         }
+        .searchable(text: $homeViewModel.searchText)
+        .onChange(of: homeViewModel.searchProduct) { newValue in
```
Comment (akabel92) [inline iStore/Home/HomeView.swift]: What's the usage of `newValue` here?

File: iStore/Home/HomeViewModel.swift [hunk 2]
```diff
@@ -28,4 +48,16 @@ final class HomeViewModel: ObservableObject {
             }
         }
     }
+    
+    func showProducts(completion: @escaping (ProductsData) -> Void) {
```
Comment (akabel92) [inline iStore/Home/HomeViewModel.swift:52]: Should be private?

File: iStore/Home/HomeView.swift (hunk 1 above)
Comment (MohamedSaiko) [reply iStore/Home/HomeView.swift]: No need for it.

File: iStore/Home/HomeViewModel.swift (hunk 2 above)
Comment (MohamedSaiko) [reply iStore/Home/HomeViewModel.swift:52]: can't be private as It's accessed from the HomeView.

getProduct() & getFilteredProduct should be private.

**Review by akabel92 — APPROVED** [review]

**Pull request merged**